From 595bafbb555260bc3e250e86db68905355bc9bbf Mon Sep 17 00:00:00 2001
From: tox <tox@tox.wtf>
Date: Tue, 14 Jul 2026 17:41:57 -0500
Subject: [PATCH] fix build against libpkgconf 30000

30000 introduced new parameters for `pkgconf_client_init()`.

Signed-off-by: tox <tox@tox.wtf>
closes #286

code consistency and confirm macro is defined

Thanks @yamader for the suggestions!

Confirm `LIBPKGCONF_VERSION` is defined before accessing it. Use
`os_get_env()`, and move preprocessor directives to the beginning of
lines to be more consistent with the rest of the codebase.

Signed-off-by: tox <tox@tox.wtf>
---
 src/external/pkgconfig_libpkgconf.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/src/external/pkgconfig_libpkgconf.c b/src/external/pkgconfig_libpkgconf.c
index ee4697fe9dcdc78a907ffee6aa9b162924fe77b8..5e3e4184b4883535a33f3df4b0fe08bb9334ae8d 100644
--- a/src/external/pkgconfig_libpkgconf.c
+++ b/src/external/pkgconfig_libpkgconf.c
@@ -18,6 +18,7 @@
 #include "log.h"
 #include "options.h"
 #include "platform/filesystem.h"
+#include "platform/os.h"
 #include "platform/path.h"
 #include "tracy.h"
 
@@ -35,12 +36,25 @@ error_handler(const char *msg, const pkgconf_client_t *client, void *data)
 	return true;
 }
 
+#if defined(LIBPKGCONF_VERSION) && LIBPKGCONF_VERSION >= 30000
+static const char *
+env_lookup_handler(const pkgconf_client_t *client, const char *key)
+{
+	(void) client;
+	return os_get_env(key);
+}
+#endif
+
 static bool
 muon_pkgconf_init(struct workspace *wk, struct pkgconf_client *c, enum machine_kind for_machine)
 {
 	TracyCZoneAutoS;
 	c->personality = pkgconf_cross_personality_default();
+#if defined(LIBPKGCONF_VERSION) && LIBPKGCONF_VERSION >= 30000
+	pkgconf_client_init(&c->client, error_handler, NULL, c->personality, NULL, env_lookup_handler);
+#else
 	pkgconf_client_init(&c->client, error_handler, NULL, c->personality);
+#endif
 
 	struct obj_array *pkg_config_path;
 	{
